File: scripts/test_user.py

import asyncio
import os
from livekit import rtc, api
from dotenv import load_dotenv

load_dotenv('.env.local')

LIVEKIT_URL = os.getenv("NEXT_PUBLIC_LIVEKIT_URL")
LIVEKIT_API_KEY = os.getenv("LIVEKIT_API_KEY")
LIVEKIT_API_SECRET = os.getenv("LIVEKIT_API_SECRET")

ROOM_NAME = 'health-help-01'
CLIENT_IDENTITY = 'Test-User-Mom-Py'

async def main():
    print(f"👤 TEST USER (Python): Connecting to {ROOM_NAME}...")
    
    # Generate Token
    grant = api.VideoGrants(room_join=True, room=ROOM_NAME)
    token = api.AccessToken(LIVEKIT_API_KEY, LIVEKIT_API_SECRET) \
        .with_grants(grant) \
        .with_identity(CLIENT_IDENTITY) \
        .with_name("Concerned Mother") \
        .to_jwt()
        
    room = rtc.Room()

    @room.on("data_received")
    def on_data_received(data: rtc.DataPacket):
        message = data.data.decode('utf-8')
        print(f"\n💬 KEITH SAYS: \"{message}\"\n")
        
        # Auto-respond logic
        if "connect with them" in message.lower():
            asyncio.create_task(delayed_response(room, "Yes, please send me the info."))

    try:
        await room.connect(LIVEKIT_URL, token)
        print("✅ User Connected.")
        
        await asyncio.sleep(2)
        await send_message(room, "Hi, I need housing for me and my young child.")
        
        # Keep alive for a bit to receive responses
        await asyncio.sleep(15)
        print("🏁 Test interactions done. Disconnecting.")
        await room.disconnect()
        
    except Exception as e:
        print(f"❌ Client Error: {e}")

async def delayed_response(room, text):
    await asyncio.sleep(2)
    await send_message(room, text)

async def send_message(room, text):
    print(f"🗣️ YOU SAY: \"{text}\"")
    data = text.encode('utf-8')
    await room.local_participant.publish_data(data, reliable=True)

if __name__ == "__main__":
    asyncio.run(main())

File: scripts/test_user.ts

import { Room, RoomEvent, RemoteParticipant, DataPacket_Kind } from 'livekit-client';
import * as dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });

// Polyfills
import 'ws';
global.WebSocket = require('ws');
global.fetch = require('cross-fetch');
// @ts-ignore
global.navigator = { userAgent: 'Node.js', product: 'ReactNative', mediaDevices: { getUserMedia: () => Promise.resolve(null), enumerateDevices: () => Promise.resolve([]) } } as any;
// @ts-ignore
global.window = { setTimeout: setTimeout, clearTimeout: clearTimeout, WebSocket: global.WebSocket, navigator: global.navigator, addEventListener: () => { }, removeEventListener: () => { } } as any;

const LIVEKIT_URL = process.env.NEXT_PUBLIC_LIVEKIT_URL || "";
const LIVEKIT_API_KEY = process.env.LIVEKIT_API_KEY || "";
const LIVEKIT_API_SECRET = process.env.LIVEKIT_API_SECRET || "";

const roomName = 'health-help-01';
const clientName = 'Test-User-Mom';

const room = new Room();

async function runClient() {
    console.log(`👤 TEST USER: Connecting to ${roomName}...`);

    try {
        await room.connect(LIVEKIT_URL, LIVEKIT_API_KEY, LIVEKIT_API_SECRET, {
            identity: clientName,
            name: "Concerned Mother",
            autoSubscribe: true,
        });
        console.log("✅ User Connected.");

        room.on(RoomEvent.DataReceived, (payload, participant) => {
            const decoder = new TextDecoder();
            const text = decoder.decode(payload);
            console.log(`\n💬 KEITH SAYS: "${text}"\n`);

            // Auto-respond for the "Agreement" phase if asked
            if (text.includes("Would you like to connect")) {
                setTimeout(() => {
                    sendMessage("Yes, please send me the info.");
                }, 2000);
            }
        });

        // Wait a moment for connection to stabilize then send first message
        setTimeout(() => {
            sendMessage("Hi, I need housing for me and my young child.");
        }, 2000);

    } catch (error) {
        console.error("❌ Client Error:", error);
    }
}

async function sendMessage(text: string) {
    console.log(`🗣️ YOU SAY: "${text}"`);
    const encoder = new TextEncoder();
    const data = encoder.encode(text);
    await room.localParticipant.publishData(data, DataPacket_Kind.RELIABLE);
}

runClient();
// Keep alive
setInterval(() => { }, 1000);
